### PR TITLE
change mutt_path_pretty() to take a Buffer

### PR DESCRIPTION
**Pull request by roccoblues**

### CODE REVIEW

File: test/path/mutt_path_pretty.c [hunk 1]
```diff
@@ -26,17 +26,40 @@
...
 {
-  // bool mutt_path_pretty(char *buf, size_t buflen, const char *homedir);
+  // bool mutt_path_pretty(struct Buffer *path, const char *homedir);
 
   {
-    TEST_CHECK(!mutt_path_pretty(NULL, 10, "apple", true));
+    TEST_CHECK(!mutt_path_pretty(NULL, "/apple", true));
   }
 
+  // test replacement
```
Comment (flatcap) [inline test/path/mutt_path_pretty.c:39]: I've looked at the [code coverage](https://neomutt.org/dev/analysis/coverage) and there are a couple of negative tests we could add:
- Path doesn't match `homedir`
- Path matches `homedir` but differs, e.g. `/home/john` vs `/home/johnsmith`

**Review by flatcap — COMMENTED** [review]
Looks good.  Thanks!

File: test/path/mutt_path_pretty.c (hunk 1 above)
Comment (roccoblues) [reply test/path/mutt_path_pretty.c:39]: Good call. Thanks for pointing me to the coverage docs. Done ✅

**Review by flatcap — APPROVED** [review]
Perfect, thanks!

**Pull request merged**